pkgs/applications/editors/vscode/generic.nix

```diff
@@ -97,4 +99,64 @@ in
     '';
 
     inherit meta;
-  }
+  };
+
+  # Vscode and variants allow for users to download and use extensions
+  # which often include the usage of pre-built binaries.
+  # This has been an on-going painpoint for many users, as
+  # a full extension update cycle has to be done through nixpkgs
+  # in order to create or update extensions.
+  # See: #83288 #91179 #73810 #41189
+  #
+  # buildFHSUserEnv allows for users to use the existing vscode
+  # extension tooling without significant pain.
+  fhs = { additionalPkgs ? pkgs: [] }: buildFHSUserEnvBubblewrap {
+    # also determines the name of the wrapped command
+    name = executableName;
+
+    # additional libraries which are commonly needed for extensions
+    targetPkgs = pkgs: (with pkgs; [
+      # ld-linux-x86-64-linux.so.2 and others
+      glibc
+
+      # dotnet
+      curl
+      icu
+      libunwind
+      libuuid
+      openssl
+      zlib
+
+      # mono
+      krb5
+    ]) ++ additionalPkgs pkgs;
+
+    # restore desktop item icons
+    extraInstallCommands = ''
+      mkdir -p $out/share/applications
+      for item in ${unwrapped}/share/applications/*.desktop; do
+        ln -s $item $out/share/applications/
+      done
+    '';
+
+    runScript = "${unwrapped}/bin/${executableName}";
+
+    # vscode likes to kill the parent so that the
+    # gui application isn't attached to the terminal session
+    dieWithParent = false;
+
+    passthru = {
+      inherit executableName;
+      inherit (unwrapped) pname version; # for home-manager module
+    };
+
+    meta = meta // {
+      description = ''
```

Comment:
I think it would be better if this description didn't replace the original, but extended it by this paragraph. Also, the descriptions of the unwrapped packages should mention that this FHS wrapped package exist, and hint at the typical kinds of bugs in which one would want to opt for it.

Reply:
> I think it would be better if this description didn't replace the original, but extended it by this paragraph.

> the descriptions of the unwrapped packages should mention that this FHS wrapped package exist

I think this would be awkward in practice, as if I extended the unwrapped version which contains a reference to the fhs version, then the fhs version would mention itself then detail how it's different.

I think this is mostly subjective. If someone was searching using https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&query=vscode , then they would find both results. And I think the current description would make the distinction clearer.

Reply:
> I think this is mostly subjective. If someone was searching using https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&query=vscode , then they would find both results. 

True, and this is also the use case for which I care. I'm imagining people clicking on `vscode`, reading the description, installing it, encountering bugs and not knowing that they should have looked at `vscode.fhs`.

Reply:
I was going to add some information to the wiki to improve discoverability anyway. I think we can open a new PR refining description if this does pose some issues for some users.